src/lib.rs

```diff
@@ -91,9 +85,42 @@ pub mod formatters;
 #[cfg(test)]
 mod test_support;
 
-/// Re-exports `log::LogLevel` so that users can initialize the emit
-/// crate without extra imports.
-pub use log::LogLevel;
+use std::fmt;
+
+#[repr(usize)]
+#[derive(Copy, Eq, Debug, PartialEq, Clone, Ord, PartialOrd)]
+pub enum LogLevel {
+    Error = 1, 
+    Warn, 
+    Info, 
+    Debug, 
+    Trace
+}
+
+#[repr(usize)]
+#[derive(Copy, Eq, Debug, PartialEq, Clone, Ord, PartialOrd)]
+pub enum LogLevelFilter {
+    Off,
+    Error, 
+    Warn, 
+    Info, 
+    Debug, 
+    Trace
+}
+
+static LOG_LEVEL_NAMES: [&'static str; 6] = ["OFF", "ERROR", "WARN", "INFO", "DEBUG", "TRACE"];
+
+impl fmt::Display for LogLevel {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        write!(fmt, "{}", LOG_LEVEL_NAMES[*self as usize])
```

Comment:
That's a neat way of stringifying an enum... I wouldn't have thought of doing that